File: iosApp/iosApp/components/SectionHeaders.swift
import SwiftUI

/// Sticky month-boundary header ("March 2022"): bold plain text directly on the background —
/// no band, no tint, no material. The opaque background fill only exists so pinned headers
/// don't let the grid scroll through the text.
struct MonthHeader: View {
    @Environment(\.colorScheme) private var scheme
    let title: String

    var body: some View {
        Text(title)
            .font(PhotosFont.monthHeader)
            .foregroundColor(PhotosColor.onSurface(scheme))
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(.horizontal, PhotosMetrics.space16)
            .padding(.top, 28)
            .padding(.bottom, PhotosMetrics.space8)
            .background(PhotosColor.background(scheme))
            .accessibilityIdentifier("timeline-month-header")
    }
}

/// Day header ("Wed, Mar 30") — the primary timeline header per Google Photos hierarchy:
/// semibold, full onSurface, plain text on the background.
struct DayHeader: View {
    @Environment(\.colorScheme) private var scheme
    let title: String

    var body: some View {
        Text(title)
            .font(PhotosFont.dateSubhead)
            .foregroundColor(PhotosColor.onSurface(scheme))
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(.horizontal, PhotosMetrics.space16)
            .padding(.top, PhotosMetrics.space20)
            .padding(.bottom, PhotosMetrics.space8)
            .background(PhotosColor.background(scheme))
            .accessibilityIdentifier("timeline-day-header")
    }
}

File: androidApp/src/main/kotlin/id/homebase/photos/android/ui/components/SectionHeaders.kt
package id.homebase.photos.android.ui.components

import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.unit.dp
import java.time.format.DateTimeFormatter

// Day-header label ("Wed, Jun 21") — UTC to match the shared month bucketing.
internal val DAY_FORMATTER: DateTimeFormatter = DateTimeFormatter.ofPattern("EEE, MMM d")

/** Month-boundary header ("March 2022") — bold plain text directly on the background, no band. */
@Composable
fun MonthHeader(
    title: String,
    modifier: Modifier = Modifier,
    testTag: String = "timeline-month-header",
) {
    Text(
        text = title,
        style = MaterialTheme.typography.headlineSmall, // monthHeader slot (Theme.kt)
        color = MaterialTheme.colorScheme.onSurface,
        modifier = modifier
            .fillMaxWidth()
            .padding(start = 16.dp, end = 16.dp, top = 28.dp, bottom = 8.dp)
            .testTag(testTag),
    )
}

/** Day header ("Wed, Mar 30") — the primary header: semibold, full onSurface, plain text. */
@Composable
fun DaySubhead(label: String, modifier: Modifier = Modifier) {
    Text(
        text = label,
        style = MaterialTheme.typography.titleSmall, // dateSubhead slot (Theme.kt)
        color = MaterialTheme.colorScheme.onSurface,
        modifier = modifier
            .fillMaxWidth()
            .padding(start = 16.dp, end = 16.dp, top = 20.dp, bottom = 8.dp)
            .testTag("timeline-day-header"),
    )
}
